service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/LaunchCreateAzureContextFlightStep.java

```diff
@@ -0,0 +1,84 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.workspace;
+
+import static bio.terra.workspace.common.utils.FlightUtils.validateRequiredEntries;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.DatabaseOperationException;
+import bio.terra.stairway.exception.FlightNotFoundException;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.model.Workspace;
+
+public class LaunchCreateAzureContextFlightStep implements Step {
+
+  private final WorkspaceService workspaceService;
+
+  public LaunchCreateAzureContextFlightStep(WorkspaceService workspaceService) {
+    this.workspaceService = workspaceService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    validateRequiredEntries(
+        context.getInputParameters(),
+        ControlledResourceKeys.SOURCE_WORKSPACE_ID,
+        JobMapKeys.AUTH_USER_INFO.getKeyName());
+    validateRequiredEntries(
+        context.getWorkingMap(), ControlledResourceKeys.CREATE_CLOUD_CONTEXT_FLIGHT_ID);
+
+    AuthenticatedUserRequest userRequest =
+        context
+            .getInputParameters()
+            .get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    Workspace destinationWorkspace =
+        context.getInputParameters().get(JobMapKeys.REQUEST.getKeyName(), Workspace.class);
+
+    String cloudContextJobId =
+        context
+            .getWorkingMap()
+            .get(ControlledResourceKeys.CREATE_CLOUD_CONTEXT_FLIGHT_ID, String.class);
+
+    boolean flightAlreadyExists;
+    try {
+      context.getStairway().getFlightState(cloudContextJobId);
+      flightAlreadyExists = true;
+    } catch (FlightNotFoundException e) {
+      flightAlreadyExists = false;
+    } catch (DatabaseOperationException e) {
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
+    }
+    // if we already have a flight, don't launch another one
```

Comment:
There could be, but I think this check is more about Stairway not repeating itself if it is going through this step for a second time which it might do if it was interrupted the first time. As far as I know, we don't need to worry about multiple WSM instances running the same flight at the same time, just about steps in a flight getting run multiple times in a row